Vagrantfile

```diff
@@ -1,19 +1,37 @@
 # -*- mode: ruby -*-
 # vi: set ft=ruby :
 
-require './lib/drupalvm/vagrant'
+require 'json'
 
-# Absolute paths on the host machine.
-host_drupalvm_dir = File.dirname(File.expand_path(__FILE__))
-host_project_dir = ENV['DRUPALVM_PROJECT_ROOT'] || host_drupalvm_dir
-host_config_dir = ENV['DRUPALVM_CONFIG_DIR'] ? "#{host_project_dir}/#{ENV['DRUPALVM_CONFIG_DIR']}" : host_project_dir
+drupalvm_env = ENV['DRUPALVM_ENV'] || 'vagrant'
 
-# Absolute paths on the guest machine.
-guest_project_dir = '/vagrant'
-guest_drupalvm_dir = ENV['DRUPALVM_DIR'] ? "/vagrant/#{ENV['DRUPALVM_DIR']}" : guest_project_dir
-guest_config_dir = ENV['DRUPALVM_CONFIG_DIR'] ? "/vagrant/#{ENV['DRUPALVM_CONFIG_DIR']}" : guest_project_dir
+# Default paths when the project is based on Drupal VM.
+host_project_dir = host_drupalvm_dir = host_config_dir = __dir__
+guest_project_dir = guest_drupalvm_dir = guest_config_dir = '/vagrant'
 
-drupalvm_env = ENV['DRUPALVM_ENV'] || 'vagrant'
+if File.exist?("#{host_project_dir}/composer.json")
+  cconfig = {}
+  composer_conf = JSON.parse(File.read("#{host_project_dir}/composer.json"))
+  if composer_conf['extra'] && composer_conf['extra']['drupalvm']
+    cconfig = composer_conf['extra']['drupalvm']
+  end
+
+  # If Drupal VM is a Composer dependency set the correct path.
+  vendor_dir = composer_conf.fetch('extra', {}).fetch('vendor-dir', 'vendor')
+  drupalvm_path = "#{vendor_dir}/geerlingguy/drupal-vm"
```

Comment:
One problem here. There are some Windows developers who might prefer to code entirely within the VM right? This is why I'm avoiding calls to `composer` without the `which` function which is now in the lib we dont have access to (d'oh).

This setup sort of works, but it still requires the user to run `composer install` to get Drupal VM. That's probably alright, I mean all window devs using this setup would have composer installed, right?

Also, as I'm avoiding calling `composer config` to retrieve the vendor-dir path, this could theoretically be wrong if the user somehow set a custom path using a different method than the `vendor-dir` extra-field.